src/esp32/tcp.rs

```diff
@@ -52,6 +57,72 @@ impl Esp32Listener {
     }
 }
 
+pub struct Esp32AsyncListener {
+    inner: Arc<TcpListener>,
+    tls: Option<Box<Esp32Tls>>,
+}
+
+impl Future for Esp32AsyncListener {
+    type Output = std::io::Result<Esp32TlsConnector>;
+    fn poll(self: std::pin::Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let stream = self.inner.accept();
+        match stream {
+            Ok(s) => {
+                let s = Esp32TlsConnector {
+                    tls: self.tls.clone(),
+                    inner: s.0,
+                };
+                Poll::Ready(Ok(s))
+            }
+            Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                cx.waker().wake_by_ref();
+                Poll::Pending
+            }
+            Err(e) => Poll::Ready(Err(e)),
+        }
+    }
+}
+
+pub struct Esp32TlsConnector {
+    tls: Option<Box<Esp32Tls>>,
+    inner: TcpStream,
+}
+
+impl Debug for Esp32TlsConnector {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Esp32TlsConnector")
+            .field("inner", &self.inner)
+            .field("tls", &self.tls.is_some())
+            .finish()
+    }
+}
+
+impl Http2Connector for Esp32TlsConnector {
+    type Stream = Esp32Stream;
+    fn accept(&mut self) -> std::io::Result<Self::Stream> {
+        match &mut self.tls {
+            Some(tls) => tls
+                .open_ssl_context(Some(self.inner.try_clone().unwrap()))
+                .map(|s| Esp32Stream::TLSStream(Box::new(s)))
+                .map_err(|e| std::io::Error::new(io::ErrorKind::Other, e)),
+            None => Ok(Esp32Stream::LocalPlain(self.inner.try_clone().unwrap())),
+        }
+    }
+}
+
+impl AsyncableTcpListener<Esp32Stream> for Esp32Listener {
+    type Output = Esp32TlsConnector;
+    fn as_async_listerner(&self) -> crate::common::conn::server::OwnedListener<Self::Output> {
```

Comment:
isn't `OwnedListener` already imported at the top of the file?

Reply:
yes!